server/chat-plugins/helptickets-auto.ts

```diff
@@ -0,0 +1,670 @@
+		if (user && !user.trusted) {
+			const result = await classifier.classify(user.name);
+			if (!result) return;
+			const keys = ['identity_attack', 'sexual_explicit', 'severe_toxicity'];
+			const matched = keys.some(k => result[k] >= 0.4);
+			if (matched) {
+				const modlog = await getModlog({
+					ip: user.latestIp,
+					actions: ['FORCERENAME', 'NAMELOCK', 'WEEKNAMELOCK'],
+				});
+				let action = determinePunishment('inapname', result, modlog);
+				if (!action) action = 'forcerename';
+				return new Map([[user.id, {
+					action,
+					user,
+					result,
+					reason: "Username detected to be breaking username rules",
+				}]]);
+			}
+		}
+	},
+	async inappokemon(ticket) {
+		const actions = new Map();
+		const links = [...getBattleLinks(ticket.text[0]), ...getBattleLinks(ticket.text[1])];
+		for (const link of links) {
+			const log = await getBattleLog(link);
+			if (!log) continue;
+			for (const [userid, pokemon] of Object.entries(log.pokemon)) {
+				let result: {
+					action: string, name: string, result: Record<string, number>, replay: string,
+				} | null = null;
+				for (const set of pokemon) {
+					if (!set.name) continue;
+					const results = await classifier.classify(set.name);
+					if (!results) continue;
+					// atm don't factor in modlog
+					const curAction = determinePunishment('inappokemon', results, []);
+					if (curAction && (!result || supersedes(curAction, result.action))) {
+						result = {action: curAction, name: set.name, result: results, replay: link};
+					}
+				}
+				if (result) {
+					// if action exists, the rest will exist. this is just for TS
+					actions.set(toID(userid), {
+						action: result.action,
+						user: toID(userid),
+						result: result.result,
+						reason: `Pokemon name detected to be breaking rules - '${result.name}'`,
+						roomid: link,
+					});
+				}
+			}
+		}
+		if (actions.size) return actions;
+	},
+	async battleharassment(ticket) {
+		const urls = getBattleLinks(ticket.text[0]);
+		const actions = new Map<string, CheckerResult>();
+		for (const url of urls) {
+			const log = await getBattleLog(url);
+			if (!log) continue;
+			const messages: Record<string, string[]> = {};
+			for (const message of log.log) {
+				const [username, text] = Utils.splitFirst(message.slice(3), '|').map(f => f.trim());
+				const id = toID(username);
+				if (!id) continue;
+				if (!messages[id]) messages[id] = [];
+				messages[id].push(text);
+			}
+			for (const [id, messageList] of Object.entries(messages)) {
+				const {averages, classified} = await getMessageAverages(messageList);
+				const punishment = determinePunishment('battleharassment', averages, []);
+				if (!punishment) continue;
+				const existingPunishment = actions.get(id);
+				if (!existingPunishment || supersedes(punishment, existingPunishment.action)) {
+					actions.set(id, {
+						action: punishment,
+						user: toID(id),
+						result: averages,
+						reason: `Not following rules in battles`,
```

Comment:
Why not incorporate the classification (sexually explicit, identity attack, rude, etc) into the reason?

Reply:
Sure, doable.